### PR TITLE
Move the builder, add the base, and nodejs-related fixes

### PR DESCRIPTION
**Pull request by aleskandro**
This PR proposes to:
- move the builder in a subfolder
- add the (forked) base image in the okd-payload-pipeline project here
- use nvm to install a specific version of nodejs, namely, the one currently used to build the openshift-console downstream.

blocks https://github.com/okd-project/okd-payload-pipeline/pull/33

### CODE REVIEW

**Review by LorbusChris — APPROVED** [review]
Not a huge fan of dl'ing nvm directly from GH, but on the other hand it seems like the best way to get the specific nodejs version that we need.

/lgtm
/approve

File: builder/Dockerfile [hunk 1]
```diff
@@ -22,11 +28,18 @@ RUN set -x; mkdir -p /go/src/ \
...
-        gcc-toolset-12 go-toolset nodejs npm openssl openssl-devel \
+        gcc-toolset-12 go-toolset openssl openssl-devel \
         systemd-devel gpgme-devel libassuan-devel \
     && yum clean all \
     # goversioninfo is not shipped as RPM in Stream9, so install it with go instead
-    && GOFLAGS='' go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@latest
+    && GOFLAGS='' go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@latest \
+    && chmod +x /usr/bin/*
+
+RUN mkdir -p "${NVM_DIR}" && curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash \
```
Comment (LorbusChris) [inline builder/Dockerfile]: this is hard to keep up to date. Any reservations against just using nvm's latest version?

Comment (aleskandro) [reply builder/Dockerfile]: no, let me update

**Comment by aleskandro** [comment]
The nodejs related fix refers https://github.com/webpack/webpack/issues/14532

File: builder/Dockerfile (hunk 1 above)
Comment (aleskandro) [reply builder/Dockerfile]: Done, merging as per discussion during our last meeting.

**Pull request merged**